int32 task_data
string task_type
int64 task_id
